src/python/pants/engine/unions.py

```diff
@@ -104,17 +104,23 @@ def __getitem__(self, union_type: _T) -> FrozenOrderedSet[_T]:
         """
         return self.union_rules[union_type]  # type: ignore[return-value]
 
-    def get(self, union_type: _T) -> FrozenOrderedSet[_T]:
+    def get(self, union_type: _T, include_subclass_members: bool = False) -> FrozenOrderedSet[_T]:
         """Get all members of this union type.
 
+        If `include_subclass_members` is True, also include all members of all direct/indirect subclasses.
+
         If the union type does not exist because it has no members registered, return an empty
         FrozenOrderedSet.
 
         Note that the type hint assumes that all union members will have subclassed the union type
         - this is only a convention and is not actually enforced. So, you may have inaccurate type
         hints.
         """
-        return self.union_rules.get(union_type, FrozenOrderedSet())  # type: ignore[return-value]
+        members = self.union_rules.get(union_type, FrozenOrderedSet())
+        if include_subclass_members:
+            for sub_union in union_type.__subclasses__():
```

Comment:
plz note this does not traverse nested hierarchies... c.f. with how I traverse the subclasses in the from_rules method.

Reply:
Ah it's only direct subclasses? :thinking: I'll add a test then change.